From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 21 Sep 2010 10:35:02 -0400
Subject: [misc] oprofile: support Intel CPU Family 6, Model 22, 29
Message-id: <1285065302-17057-1-git-send-email-jolsa@redhat.com>
Patchwork-id: 28325
O-Subject: [PATCH RHEL5] 493047 oprofile: add support for Intel CPU Family 6 /
	Model 22, 29
Bugzilla: 493047
RH-Acked-by: Don Zickus <dzickus@redhat.com>

Bugzilla: 493047
https://bugzilla.redhat.com/show_bug.cgi?id=493047

Description:
============
BZ is asking for Model 29 support. Adding also Model 22 to be in sync
with upstream.

Upstream status:
================
- Add Support for Intel CPU Family 6 / Model 22 (Intel Celeron 540)
  commit c4c3dd68d9075783ea5f1cfe268446c80e4ebf38
  Author: Patrick Simmons <linuxrocks123@netscape.net>

- Add Support for Intel CPU Family 6 / Model 29
  http://lkml.indiana.edu/hypermail/linux/kernel/1009.2/02028.html
  acked, on its way to upstream

Brew:
=====
https://brewweb.devel.redhat.com/taskinfo?taskID=2754301

Tested:
=======
tested with oprofile, CPU correctly recognized

wbr,
jirka

diff --git a/arch/i386/oprofile/nmi_int.c b/arch/i386/oprofile/nmi_int.c
index 9307d94..24922c0 100644
--- a/arch/i386/oprofile/nmi_int.c
+++ b/arch/i386/oprofile/nmi_int.c
@@ -442,7 +442,10 @@ static int __init ppro_init(char **cpu_type)
 	case 14:
 		*cpu_type = "i386/core";
 		break;
-	case 15: case 23:
+	case 0x0f:
+	case 0x16:
+	case 0x17:
+	case 0x1d:
 		*cpu_type = "i386/core_2";
 		break;
 	case 0x2e:
